pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -393,11 +388,32 @@ func (m *MultishareOpsManager) runEligibleInstanceCheck(ctx context.Context, req
 		}
 
 		klog.Infof("Instance %s/%s/%s with state %s is not ready with ongoing operation %s type %s", instance.Project, instance.Location, instance.Name, instance.State, op.Id, op.Type.String())
-		nonReadyInstanceCount += 1
+		nonReadyEligibleInstances = append(nonReadyEligibleInstances, instance)
+
 		// TODO: If we see > 1 instances with 0 shares (these could be possibly leaked instances where the driver hit timeout during creation op was in progress), should we trigger delete op for such instances? Possibly yes. Given that instance create/delete and share create/delete is serialized, maybe yes.
 	}
 
-	return readyEligibleInstances, nonReadyInstanceCount, nil
+	if len(readyEligibleInstances) == 0 && len(nonReadyEligibleInstances) > 0 {
+		errorString := "All eligible filestore instances are busy.\n"
+
+		for _, instance := range nonReadyEligibleInstances {
+			op, err := containsOpWithInstanceTargetPrefix(instance, ops) // Error for this call is already checked above
+			if err != nil {
+				klog.Errorf("failed to check eligibility of instance %s", instance.Name)
+				return nil, err
```

Comment:
by the time we have reached line 396, we have already determined the `readyEligibleInstances` and `nonReadyEligibleInstances`. Can we just log a kWarning here and continue?

Reply:
If we log a klog warning, the user won't be able to see it in their events.  

The if statement here is to keep the same decision tree as we had before. Before the caller method would check if there are no ready instances, and nonReady was >0 it would return the error event for user to see. (Previously line 131)

I didn't want to change the actual logic of when we show the error.

Reply:
Ack